user-query-optimizer/optimizations/approximates.py

```diff
@@ -0,0 +1,39 @@
+import re
+import sqlparse
+from collections import defaultdict
+from collections import OrderedDict
+from clickhouse_cli.ui.parseutils.ctes import extract_ctes
+from sqlparse.sql import IdentifierList, Identifier, Function, Where, Comparison
+from sqlparse.tokens import Keyword, DML, Newline, CTE, Wildcard
+
+# Optimization #1
```

Comment:
Consider how to formalize these explanations by using ReadTheDocs [0]. That would auto-build some documentation that we can use to explain e.g. what the different optimizations are, how to use the Optimizer class, etc.

[0] https://docs.readthedocs.io/en/latest/getting_started.html